massdriver/resource_package_alarm.go

```diff
@@ -80,6 +83,24 @@ func resourcePackageAlarm() *schema.Resource {
 					},
 				},
 			},
+			"threshold": {
+				Description: "The threshold for triggerin the alarm",
+				Type:        schema.TypeFloat,
+				ForceNew:    true,
+				Optional:    true,
+			},
+			"period_minutes": {
+				Description: "The number of periods over which data is compared to the specified threshold",
+				Type:        schema.TypeInt,
+				ForceNew:    true,
+				Optional:    false,
```

Comment:
If you intend to make these required, you need to use `Required: true` rather than `Optional: false`. [I don't think `Optional: false` is valid](https://github.com/hashicorp/terraform-plugin-framework/issues/31). Did this update pass validation?

Also, keep in mind that if you add new parameters and make them required, then *every* existing use of this resource will immediately be broken upon updating to this version (which includes most of our bundles).

I noticed there is an [update to our module](https://github.com/massdriver-cloud/terraform-modules/pull/194/files) which gets around this ^, but both of these changes independently break most of our bundles. A new provider will have missing fields, a updated module will be setting non-existent values. We will have to coordinate a simultaneous release of:
* this provider + updating the provisioner image
* merging the terraform module change
* updating every one of our bundles that uses that module to bump the module ref

All that is to say ^. Do these fields really need to be required (right now)? Or can we roll this out slowly without it and add it later (if ever).

Reply:
Meant to mark these all as optional, my fault.